### ISSUE TITLE
Negative loss and IoU is greater than 1

### ISSUE THREAD

**Issue by numancelik34** [qubvel/segmentation_models.pytorch#256]
I am getting a weird loss and IoU score.. Dice loss becomes negative just after 2-3 epochs, and IoU becomes greater than 1 at the same time.
Why is this happening? I have already checked and my masks are binary images in {0, 1} values.
Something wrong in loss function??

Thanks

**Comment by cristophersfr** [comment]
I'm not an expert, but I would avise you would check your activation functions in particular. Probably your output is not being properly matched to your masks. Nonetheless, your optimizer gets lost and start "diverging".

**Comment by AlessandroSaviolo** [comment]
> I am getting a weird loss and IoU score.. Dice loss becomes negative just after 2-3 epochs, and IoU becomes greater than 1 at the same time.
> Why is this happening? I have already checked and my masks are binary images in {0, 1} values.
> Something wrong in loss function??
> 
> Thanks

Hi! Did you solve the problem? I'm experiencing the same weird behaviour

**Comment by cristophersfr** [comment]
This is probably a problem with your loss function. You should check your loss with both the same input and output, if it is zero, then your problem is much likely your tensors.

**Comment by AlessandroSaviolo** [comment]
Thank you for the quick reply! I may have found the problem. I was not normalizing the images/masks to 0-1
